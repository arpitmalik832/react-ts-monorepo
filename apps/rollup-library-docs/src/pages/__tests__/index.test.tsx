import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Home from '../index';

jest.mock('../../components/HomepageFeatures', () => () => (
  <div data-testid="mock-home-features">HomepageFeatures</div>
));

describe('Home unit tests', () => {
  it('snapshot test', () => {
    const component = render(<Home />);

    expect(component).toMatchSnapshot();
  });
});
