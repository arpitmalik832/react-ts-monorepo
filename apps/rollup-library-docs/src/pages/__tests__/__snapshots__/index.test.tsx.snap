// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home unit tests snapshot test 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="mock-layout"
      >
        <div
          data-testid="mock-layout-title"
        >
          Hello from mock title
        </div>
        <div
          data-testid="mock-layout-description"
        >
          Description will go into a meta tag in &lt;head /&gt;
        </div>
        <div
          data-testid="mock-layout-children"
        >
          <header
            class="hero hero--primary heroBanner"
          >
            <div
              class="container"
            >
              <h1
                class="hero__title"
              >
                mock title
              </h1>
              <p
                class="hero__subtitle"
              >
                mock tagline
              </p>
              <div
                class="buttons"
              >
                <div
                  data-testid="mock-link"
                >
                  <div
                    data-testid="mock-link-classname"
                  >
                    button button--secondary button--lg
                  </div>
                  <div
                    data-testid="mock-link-to"
                  >
                    /docs/intro
                  </div>
                </div>
              </div>
            </div>
          </header>
          <main>
            <div
              data-testid="mock-home-features"
            >
              HomepageFeatures
            </div>
          </main>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="mock-layout"
    >
      <div
        data-testid="mock-layout-title"
      >
        Hello from mock title
      </div>
      <div
        data-testid="mock-layout-description"
      >
        Description will go into a meta tag in &lt;head /&gt;
      </div>
      <div
        data-testid="mock-layout-children"
      >
        <header
          class="hero hero--primary heroBanner"
        >
          <div
            class="container"
          >
            <h1
              class="hero__title"
            >
              mock title
            </h1>
            <p
              class="hero__subtitle"
            >
              mock tagline
            </p>
            <div
              class="buttons"
            >
              <div
                data-testid="mock-link"
              >
                <div
                  data-testid="mock-link-classname"
                >
                  button button--secondary button--lg
                </div>
                <div
                  data-testid="mock-link-to"
                >
                  /docs/intro
                </div>
              </div>
            </div>
          </div>
        </header>
        <main>
          <div
            data-testid="mock-home-features"
          >
            HomepageFeatures
          </div>
        </main>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
